app/src/main/java/com/amaze/filemanager/utils/files/FileUtils.java

```diff
@@ -662,17 +682,19 @@ public static boolean canGoBack(Context context, HybridFile currentFile) {
         }
     }
 
-    public static long[] getSpaces(HybridFile hFile, Context context, final OnProgressUpdate<Long[]> updateState) {
+    public static Long[] getSpaces(HybridFile hFile, Context context, final OnProgressUpdate<Long[]> updateState) {
         long totalSpace = hFile.getTotal(context);
         long freeSpace = hFile.getUsableSpace();
         long fileSize = 0l;
+        Long[] spaces = new Long[] {totalSpace, freeSpace, fileSize};  
 
         if (hFile.isDirectory(context)) {
-            fileSize = hFile.folderSize(context);
+            fileSize = hFile.folderSize(context, updateState, spaces);  
```

Comment:
That's because we need the whole array to  updateState.onUpdate in the same file.

    public static long folderSize(File directory, final OnProgressUpdate<Long[]> updateState, Long[] spaces) {    
...
        if(updateState != null)
            updateState.onUpdate(spaces);